code/modules/mob/mob_movement.dm

```diff
@@ -76,8 +57,9 @@
 		return FALSE
 	if(HAS_TRAIT(mob, TRAIT_NO_TRANSFORM))
 		return FALSE //This is sorta the goto stop mobs from moving trait
-	if(mob.control_object)
-		return Move_object(direct)
+	if(HAS_TRAIT(mob, TRAIT_CURRENTLY_CONTROLLING_OBJECT)) // different signal than COMSIG_MOB_CLIENT_PRE_LIVING_MOVE for distinction when it comes to move code's bone hurting juice
+		SEND_SIGNAL(mob, COMSIG_MOB_CLIENT_MOVE_POSSESSED_OBJECT, new_loc, direct)
+		return FALSE
```

Comment:
Why not just add another generic signal - in the `!isliving` conditional - `COMSIG_MOB_CLIENT_PRE_GHOST_MOVE`.